centreon/www/class/centreonDowntime.class.php

```diff
@@ -265,211 +265,319 @@ public function getInfos($id)
     }
 
     /**
-     * Get the list of relations for a downtime
+     * Intends to return hosts, hostgroups, services, servicesgroups linked to the recurrent downtime
      *
-     * <code>
-     * $return_array =
-     *  array(
-     *      'host' => array, // The list of host id
-     *      'hostgrp' => array, // The list of hostgroup id
-     *      'svc' => array, // The list of service id
-     *      'svcgrp' => array, // The list of servicegroup id
-     *  )
-     * </code>
-     *
-     * @param int $id The downtime id
-     * @return array The list of relations
+     * @param integer $downtimeId
+     * @return array
      */
-    public function getRelations($id)
+    public function getRelations(int $downtimeId): array
     {
-        $list = array(
-            "host" => array(),
-            "hostgrp" => array(),
-            "svc" => array(),
-            "svcgrp" => array()
-        );
-        foreach (array_keys($list) as $type) {
-            switch ($type) {
-                case 'host':
-                    $query = "SELECT host_host_id as obj_id FROM downtime_host_relation WHERE dt_id = :id";
+        $relations = [
+            'hosts' => [],
+            'hostgroups' => [],
+            'services' => [],
+            'servicegroups' => [],
+        ];
+
+        foreach (array_keys($relations) as $resourceType) {
+            switch ($resourceType) {
+                case 'hosts':
+                    $query = 'SELECT
+                        dhr.host_host_id AS resource_id,
+                        h.host_activate AS activated
+                    FROM downtime_host_relation dhr
+                    INNER JOIN host h
+                        ON dhr.host_host_id = h.host_id
+                    WHERE
+                        dt_id = :downtimeId';
```

Comment:
To fix indentation inside the used string, you can use HEREDOC syntax (note the additionnal indentation needed by php-cs-fixer).
```suggestion
                    $query = <<<'SQL'
                        SELECT
                            dhr.host_host_id AS resource_id,
                            h.host_activate AS activated
                        FROM downtime_host_relation dhr
                        INNER JOIN host h
                            ON dhr.host_host_id = h.host_id
                        WHERE
                            dt_id = :downtimeId';
                        SQL;
```